src/parser/shared/modules/CooldownThroughputTracker.js

```diff
@@ -52,10 +53,19 @@ class CooldownThroughputTracker extends Analyzer {
   addCooldown(cooldownSpell, timestamp) {
     const cooldown = {
       ...cooldownSpell,
-      start: timestamp,
+      start: timestamp - (cooldownSpell.startBufferMS || 0),
       end: null,
       events: [],
     };
+    if (cooldown.startBufferMS) {
+      const combatantCasts = this.owner.eventHistory.filter(event => event.type === EventType.Cast && event.sourceID === this.selectedCombatant._combatantInfo.sourceID);
+      const preCasts = combatantCasts.filter(event => event.timestamp >= timestamp - cooldown.startBufferMS && event.timestamp <= timestamp);
+      preCasts.reverse().forEach(event => cooldown.events.unshift(event));
+    } else if (cooldown.startBufferSpells) {
+      const combatantCasts = this.owner.eventHistory.filter(event => event.type === EventType.Cast && event.sourceID === this.selectedCombatant._combatantInfo.sourceID);
```

Comment:
If you wanted to go all out on reducing duplicate code, you could do something like this

    if (cooldown.startBufferMS || cooldown.startBufferSpells) {
        const combatantCasts = this.owner.eventHistory.filter(event => event.type === EventType.Cast && event.sourceID === this.selectedCombatant._combatantInfo.sourceID);

        if (cooldown.startBufferMS) {
            const preCasts...
        } else if (cooldown.startBufferSpells) {
            const preCasts...
        }
    }

Not really a big deal so I'll set this to approved and unmerged and leave it up to you.

Reply:
Yea i was looking at reducing it into something a bit shorter ... similar to what you did, maybe ill throw that in